Epona/include/Epona/Analysis.hpp

```diff
@@ -158,71 +168,126 @@ void FindExtremalVertices(
 }
 
 /**
- *  @brief  Calculates box vertices in the world coordinate space from a given
- *          orthogonal basis and it's position
+ * @brief Calculates vertex position in the world space
  *
- *  Writes output vertices to the container starting with @p verticesBeginIterator
+ * @tparam  Vector the vector type that has overloaded vector-vector addition operator
+ * @tparam  Matrix the matrix type that has overloaded matrix-vector multiplication operator
  *
- *  @attention  There must be at least 7 more elements following given iterator
+ * @param   vertex      the vertex data
+ * @param   translation the vertex translation vector
+ * @param   rotation    the vertex rotation matrix
+ * @param vertex in the world coordinates
+ */
+template < typename Vector, typename Matrix >
+constexpr Vector ModelToWorldSpace(
+        Vector vertex, Vector translation, Matrix rotation
+    )
+{
+    return rotation * vertex + translation;
+}
+
+/**
+ * @brief Calculates vertex position in the model space
  *
- *  @tparam VerticesContainerIt Random access iterator
+ * @tparam  Vector the vector type that has overloaded vector-vector subtraction operator
+ * @tparam  Matrix the matrix type that has overloaded matrix-vector multiplication operator
  *
- *  @param[in]  i       box axis vector
- *  @param[in]  j       box axis vector
- *  @param[in]  k       box axis vector
- *  @param[in]  center  center of the box
- *  @param[out] verticesBeginIterator   iterator to the container
+ * @param   vertex           the vertex data
+ * @param   translation      the vertex translation vector
+ * @param   inverseRotation  the vertex inverse rotation matrix
```

Comment:
In `ModelToWorldSpace` we supply `translation` and `rotation` but here we supply `translation` but `inverse rotation`? Is there a specific reason not to supply the same `rotation`?